.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "deps/hiredis"]
+	path = deps/hiredis
+	url = https://github.com/redis/hiredis.git
```

Comment:
Maybe not part of this PR but do you think we can use subtree instead of submodule? With subtree, dependency project code is always in the repo, no need to run extra commands to fetch/clear it. (As I mentioned before, I also feel the same way for cmocka as well, it should be part of the repo to make the developers' life easier)

Reply:
I am not much familiar with subtree. Sound like a good idea. Let's do it on a distinct PR.